state.go

```diff
@@ -1085,8 +1095,19 @@ func LoadStateData(store Store) (StateData, error) {
 	}
 
 	var sd StateData
+	// we are relying on the fact that Unmarshal will decode all and only the fields
+	// that it can find in the destination type.
 	err = json.Unmarshal(data, &sd)
-	return sd, err
+	if err != nil {
+		return StateData{}, err
+	}
+
+	switch sd.Version {
+	case 0, 1:
+		return sd, nil
+	default:
+		return sd, errors.New("unsupported state data version")
```

Comment:
Maybe. Will fix. The idea was that even though you have some version which is not supported, maybe you will be able to get some data from it anyway.
But I agree, that for now this is not needed.